### ISSUE TITLE
Disabling ublocklite doesn't fully disable it 

### ISSUE THREAD

**Issue by Clashpoint** [uBlockOrigin/uBOL-issues#20]
I am trying to disable ublock completely on some website so I see ads, an example of a website I need the ad block to fully ignore is `tab.gladly.io` (tab for a cause) , the issue when I fully disable the filters (the extension clearly says no filters and after reloading) it still wouldn't show ads at all .

**Comment by gorhill** [comment]
You need to provide more information if you want me to reproduce on my side, I can't investigate what I can't reproduce. Tell me everything I need to do in details for me to reproduce -- keep in mind I am not behind you looking at your monitor.

**Comment by Clashpoint** [comment]
that's fair, I apologize for not being clear , let me know if any more info is needed 
I am using google chrome version Version 108.0.5359.125 (Official Build) (64-bit)
while testing this I have only _ublock origin lite_ and _tab for a cause_ as my enabled extensions 
these are the errors that I have in my console 
![chrome_QmByPe4Xzl](https://user-images.githubusercontent.com/28152443/208479949-cc8b2553-e678-4b0d-8352-c96a1f6dfa6c.png)

and this is whay my ublock shows 
![kQwgbn1z9O](https://user-images.githubusercontent.com/28152443/208480058-bc80ea12-8443-43c5-b3f1-07c10f69996f.png)
page with ublock as an active extension with filters fully disabled 
![chrome_3uXRbSXtIB](https://user-images.githubusercontent.com/28152443/208480170-cdd8d9d4-8ae5-4f1b-b452-d3b37fa6a03b.png)
and the page when it's disabled as an extension 
![chrome_RjND67C8mC](https://user-images.githubusercontent.com/28152443/208480397-4a2b0a93-169b-44f5-86f4-c6368edf68d2.png)
tho even while disabled I am getting these errors in my console 
![chrome_w2Q1mYfrNg](https://user-images.githubusercontent.com/28152443/208480601-7ea0e53c-79f6-48fb-b41e-f2fd5df558d3.png)

**Comment by gorhill** [comment]
What's your default filtering mode in uBO Lite?

**Comment by Clashpoint** [comment]
default filtering mode  is optimal with these filters 
![chrome_7TPjmfALQx](https://user-images.githubusercontent.com/28152443/208484998-8c866729-ac90-49f7-8f3d-834f20d2de1b.png)

**Comment by gorhill** [comment]
The problem does not occur when default mode is set to _Basic_.

Unfortunately, I do not know how this can be fixed, as there is no way to tell the [scripting API](https://developer.chrome.com/docs/extensions/reference/scripting/) to declaratively exclude script injection according to the root context. Another limitation is that with the [permissions API](https://developer.chrome.com/docs/extensions/reference/permissions/) there is no way to exclude a specific site from an extension.

For now the only workaround I see is to work in _Basic_ mode by default, and enable higher blocking modes only for specific sites.

**Comment by Clashpoint** [comment]
I see, thanks for the reply

**Comment by erock8303** [comment]
This issue is affecting other extensions from being used on specific sites for me and my team.

We use Hubspot for our sales emails to track and log them in our CRM. Their Hubspot Sales Extension allows us to do this within the Gmail interface. Unfortunately, uBOL does not completely "turn off" when I select 'no filtering' on mail.google.com. The Hubspot extension fails every time uBOL is installed. It works when uBOL is uninstalled. uBOL is our companies only allowed adblocker.

I've worked with Hubspot support team and have exhausted all options to work around uBOL. They provided a list of their domains that I added to uBOL's "allow" list, but no change to the failing functionality.

Any suggestions?

Let me know if you need any more info, screenshots, or anything else. We would really appreciate your help in solving this issue.

**Comment by gorhill** [comment]
Following the discussion at <https://github.com/w3c/webextensions/issues/668>, I did try the suggestion to use a [synchronous XHR-based hacky solution](https://github.com/w3c/webextensions/issues/668#issuecomment-2263769391), and I have a prototype which solves the issue. It works this way:

Create a dynamic DNR rule as follow

```js
{
    id: ALLOWALL_HACK,
    action: {
        type: 'redirect',
        redirect: {
            extensionPath: '/web_accessible_resources/filtering-enabled'
        },
    },
    condition: {
        resourceTypes: [ 'xmlhttprequest' ],
        urlFilter: '|https://ubol.invalid/filtering/',
    },
    priority: 100,
}
```

`priority` is set to the same `priority` used to declare the `allowAllRequests` rules used to implement the no-filtering mode, hence only those `allowAllRequests` can bypass the `ALLOWALL_HACK` rule.

Content script side, there is this code:

```js
// https://github.com/uBlockOrigin/uBOL-home/issues/20
// https://github.com/w3c/webextensions/issues/668
// Hacky, until a better solution is possible in MV3.
// If the xhr fails, this means the request was not redirected to a local
// resource by the extension, which indicates that no-filtering mode is in
// effect for the root context.
{
    const ancestors = document.location.ancestorOrigins;
    if ( ancestors && ancestors.length !== 0 ) {
        const rootOrigin = ancestors.item(ancestors.length-1);
        if ( rootOrigin !== document.location.origin ) {
            const xhr = new XMLHttpRequest();
            try {
                xhr.open('GET', `https://ubol.invalid/filtering/`, false);
                xhr.send();
            } catch(ex) {
                return; // no filtering should occur
            }
        }
    }
}
```
So when disabling filtering for `example.com`, an `allowAllRequests` rule is created for `example.com`, which will cause the `ALLOWALL_HACK` rule to be disabled when `example.com` is one of the ancestor, and the content script will return immediately as a result.

However this comes at the cost of undesirable side effects:

`https://ubol.invalid/filtering/` is not secret, so websites could use it to find out whether uBO is installed and enabled. A per-release random part could be used but the URL would still be public and usable for the duration of the release

With sites for which uBOL is disabled, the console may be polluted by numerous error messages because `https://ubol.invalid/filtering/` is unreachable:

![image](https://github.com/user-attachments/assets/34f9f56f-4649-4274-97a5-8ac6b881a084)

**Comment by stephenhawk8054** [comment]
> `https://ubol.invalid/filtering/` is not secret, so websites could use it to find out whether uBOL is installed and enabled. A per-release random part could be used but the URL would still be public and usable for the duration of the release

Hmm... That sounds like a trivial way for websites' anti-adblock?

**Comment by gorhill** [comment]
> That sounds like a trivial way for websites' anti-adblock?

Yes, hence why it's unlikely I go ahead with this fix, it was just a proof of concept that this can be done, but there is a cost. The real solution would be for uBOL to be able to dynamically register data to be injected in isolated and main world and accessible to registered scripts.

**Comment by gorhill** [comment]
Another approach is to use `storage.session()` to retrieve hostnames for no-filtering sites, then use the result to revert the injection. `storage.session()` can be available from injected scripts with [`TRUSTED_CONTEXTS` access level](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/StorageArea/setAccessLevel). This wouldn't work for scriptlets though as these must execute immediately and there is no way to revert these.

**Comment by SuihtilCod** [comment]
I'm experiencing this as well.

Under *uBlock Origin Lite* (version 2025.1104.1637), I set filtering to "disabled" for a website I enjoy — so that their non-intrusive ads could play.  However, even with filtering set to "disabled," the extension **still blocks the ads.**

#### Additional Info

- The specific ads are being blocked exclusively by **EasyList**.
- The issue persists despite trying to disable filtering for this site.
- Disabling the extension altogether or disabling EasyList both allow the ads to show.

**Comment by u-RraaLL** [comment]
https://www.reddit.com/r/uBlockOrigin/comments/1spd89p/comment/ogzu27n/
